lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -173,16 +173,48 @@ def relationship_value_for(serializer, options = {})
           end
         end
 
+        def relationship_links_for(name, owner, options = {})
+          options[:links].each_with_object({}) do |(key, link), hash|
+            if link.is_a? Proc
+              hash[key] = link.call(owner, name)
+            else
+              hash[key] = link
+            end
+
+            hash
+          end
+        end
+
+        def relationship_meta_for(name, owner, options = {})
+          if options[:meta].is_a? Proc
+            options[:meta].call(owner, name)
+          else
+            options[:meta]
+          end
+        end
+
         def relationships_for(serializer)
           serializer.associations.each_with_object({}) do |association, hash|
-            hash[association.key] = { data: relationship_value_for(association.serializer, association.options) }
+            hash[association.key] = {}
+
+            if association.options.fetch(:data, true)
```

Comment:
`association.options.fetch(:data, true)` is called a lot, is feature envy, seems association should have an accessor `association.data` `association.data=(data)` or perhaps something better named